crates/bitcoin-da/src/helpers/builders.rs

```diff
@@ -336,34 +336,76 @@ pub fn create_inscription_transactions(
     reveal_fee_rate: f64,
     network: Network,
     reveal_tx_prefix: &[u8],
+) -> Result<(Transaction, TxWithId), anyhow::Error> {
+    // Assume body size < 400kb:
+    create_inscription_type_0(
+        rollup_name.as_bytes(),
+        body,
+        signature,
+        sequencer_public_key,
+        prev_tx,
+        utxos,
+        recipient,
+        reveal_value,
+        commit_fee_rate,
+        reveal_fee_rate,
+        network,
+        reveal_tx_prefix,
+    )
+}
+
+// TODO: parametrize hardness
+// so tests are easier
+// Creates the inscription transactions (commit and reveal) Type 0
+#[allow(clippy::too_many_arguments)]
+#[instrument(level = "trace", skip_all, err)]
+pub fn create_inscription_type_0(
+    rollup_name: &[u8],
+    body: Vec<u8>,
+    signature: Vec<u8>,
+    sequencer_public_key: Vec<u8>,
```

Comment:
can you call this signer_public_key?

bc prover also uses this method. since we are refactoring here, we can rename